.github/workflows/apix-ci.yml

```
@@ -0,0 +1,109 @@
+      - packages/extension-api-explorer/**
+
+  push:
+    branches:
+      - main
+    paths:
+      - packages/code-editor/**
+      - packages/run-it/**
+      - packages/api-explorer/**
+      - packages/extension-api-explorer/**
+
+  workflow_dispatch:
+
+env:
+  TS_JUNIT_OUTPUT_DIR: results/apix
+
+jobs:
+  unit:
+    name: Unit - ${{ matrix.os }} / Node ${{ matrix.node-version }}
+    env:
+      JEST_JUNIT_OUTPUT_DIR: results/apix
+      JEST_JUNIT_OUTPUT_NAME: ${{ matrix.os }}.tssdkrtl${{ matrix.node-version }}.xml
+    runs-on: ${{ matrix.os }}-latest
+
+    strategy:
+      matrix:
+        os:
+          - ubuntu
+        node-version: [12.x]
+
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Repo Checkout
+        uses: actions/checkout@v2
+
+      - name: Install Node ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Install dependencies
+        run: |
+          yarn
+          yarn lint:es --quiet
+          yarn build
+
+      - name: Run unit tests
+        run: yarn jest "packages/(code-editor|run-it|api-explorer|extension-api-explorer)/src" --reporters=default --reporters=jest-junit
+
+      - name: Upload ts unit test results
+        if: ${{ always() }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: apix-test-results
+          path: results/apix
+
+  publish-test-results:
+    needs: [unit]
+    if: success() || failure()
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Download Artifacts
+        uses: actions/download-artifact@v2
+        with:
+          path: artifacts
+
+      - name: Publish Unit Test Results
+        uses: EnricoMi/publish-unit-test-result-action@v1.12
+        with:
+          check_name: Typescript Tests
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          report_individual_runs: true
+          hide_comments: orphaned commits
+          check_run_annotations_branch: '*'
+          files: 'artifacts/apix-test-results/*.xml'
+
+  # The following jobs are necessary because they are configured as required
+  # checks on PRs and would not otherwise run in this workflow. If the PR/branch
+  # triggering this workflow file also needs to actually run these tests, the
+  # other workflow files will pick it up and submit the results. In that case
+  # there will be duplicate PR check names but github handles this and requires
+  # both to be passing.
+  noop-codegen-results:
+    name: Codegen Tests
+    needs: [publish-test-results]
+    if: success() || failure()
+    runs-on: ubuntu-latest
+    steps:
+      - run: exit 0
+
+  noop-python-results:
+    name: Python Tests
+    needs: [publish-test-results]
+    if: success() || failure()
+    runs-on: ubuntu-latest
+    steps:
+      - run: exit 0
```

```suggestion
      - run: exit 0

  noop-typescript-results:
    name: Typescript Tests
    needs: [publish-test-results]
    if: success() || failure()
    runs-on: ubuntu-latest
    steps:
      - run: exit 0
```

These noop tests are annoying but our best option to have ["required checks"](https://github.community/t/github-actions-and-required-checks-in-a-monorepo-using-paths-to-limit-execution/16586/2)  (which is still actually a work in progress)

Which means we also need the following block added to the end of these files:
* .github/workflows/codegen-ci.yml
* .github/workflows/python-ci.yml
* .github/workflows/tssdk-ci.yml

```yaml
  noop-typescript-results:
    name: APIX Tests
    needs: [publish-test-results]
    if: success() || failure()
    runs-on: ubuntu-latest
    steps:
      - run: exit 0
```

---

ug, scratch all this - gonna go with a different approach to "required checks"